### PR TITLE
ORC-946: Unified json library

### PR DESCRIPTION
**Pull request by guiyanakuang**
### What changes were proposed in this pull request?

The java project depends on several json libraries. Replacing jackson/jettison with Gson 

```
use jackson-core in orc-benchmarks-core
org.apache.orc.bench.core.convert.json.JsonWriter

use jettison in tools
org.apache.orc.tools.KeyTool
org.apache.orc.tools.JsonFileDump
org.apache.orc.tools.PrintData
```

gson vs. jettison has some inconsistent

1.  the scope of character escaping is inconsistent; 
    jettison escapes '/'
    gson does not modify
2. Inconsistent floating point writing; 
    jettison removes trailing zeros and decimal points where possible. 
    gson leaves it as is.
3. prettyPrint;
    jettison's prettyPrint is faulty, '[' and '{' are often linked together without a line break. 
    gson prettyPrint is fine.

These three behaviours are not forward compatible in this pr.

I don't think prettyPrint needs to be made compatible. The other two I would like to hear your opinion on.

### Why are the changes needed?

Reduce dependencies and use a unified json lib.

### How was this patch tested?

Pass the CIs.

### CODE REVIEW

**Comment by dongjoon-hyun** [comment]
Thank you for making a PR, @guiyanakuang .

File: java/tools/src/test/resources/orc-file-dump.json [hunk 1]
```diff
@@ -1323,46 +1331,52 @@
...
+          }
         }
-      }]
+      ]
     }
   ],
   "fileLength": 272513,
   "paddingLength": 0,
-  "paddingRatio": 0,
+  "paddingRatio": 0.0,
```
Comment (dongjoon-hyun) [inline java/tools/src/test/resources/orc-file-dump.json:1380]: I agree that new floating point string looks better.

File: java/tools/src/test/resources/orc-file-dump.json [hunk 2]
```diff
@@ -28,7 +28,7 @@
       }
     }
   },
-  "calendar": "Julian\/Gregorian",
+  "calendar": "Julian/Gregorian",
```
Comment (dongjoon-hyun) [inline java/tools/src/test/resources/orc-file-dump.json:31]: This one looks good too.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Thank you, @guiyanakuang .
Merged to main for Apache ORC 1.8.

**Comment by guiyanakuang** [comment]
Thanks to dongjoon-hyun for the review and approval!

**Pull request merged**